src/modules/aws_codedeploy/rpcs/start_deploy.ts

```diff
@@ -0,0 +1,218 @@
+        },
+      ];
+    }
+    if (!revision) {
+      return [
+        {
+          id: '',
+          status: 'KO',
+          message: 'Please provide the specification of the RevisionLocation to use',
+        },
+      ];
+    }
+    if (!region) {
+      return [
+        {
+          id: '',
+          status: 'KO',
+          message: 'Please provide the region of the CodeDeploy application to deploy',
+        },
+      ];
+    }
+
+    // validate if application name exists
+    const appObj =
+      (await this.module.application.db.read(
+        ctx,
+        this.module.application.generateId({ name: applicationName, region }),
+      )) ??
+      (await this.module.application.cloud.read(
+        ctx,
+        this.module.application.generateId({ name: applicationName, region }),
+      ));
+
+    if (!appObj) {
+      return [
+        {
+          id: '',
+          status: 'KO',
+          message: 'CodeDeploy application not found',
+        },
+      ];
+    }
+
+    // validate if deployment group name exists
+    if (deploymentGroupName) {
+      const dgObj =
+        (await this.module.deploymentGroup.db.read(
+          ctx,
+          this.module.deploymentGroup.generateId({ deploymentGroupName, applicationName, region }),
+        )) ??
+        (await this.module.deploymentGroup.cloud.read(
+          ctx,
+          this.module.deploymentGroup.generateId({ deploymentGroupName, applicationName, region }),
+        ));
+
+      if (!dgObj) {
+        return [
+          {
+            id: '',
+            status: 'KO',
+            message: 'CodeDeploy deployment group not found',
+          },
+        ];
+      }
+    }
+
+    const client = (await ctx.getAwsClient(region)) as AWS;
+    const revisionObj: RevisionLocation = JSON.parse(revision);
+    const input: CreateDeploymentCommandInput = {
+      applicationName,
+      deploymentGroupName,
+      revision: revisionObj,
+    };
+    const deploymentId = await this.startDeploy(client.cdClient, input);
+
+    if (!deploymentId) {
+      return [
+        {
+          id: '',
+          status: 'KO',
+          message: 'Error creating deployment',
+        },
+      ];
+    }
+
+    // wait until deployment is succeeded
+    const result = await waitUntilDeploymentSuccessful(
+      {
+        client: client.cdClient,
+        // all in seconds
+        maxWaitTime: 1200,
+        minDelay: 1,
+        maxDelay: 4,
+      } as WaiterOptions<CodeDeploy>,
+      { deploymentId },
+    );
+
+    // update
+    const dbDeploy = await this.module.deployment.deploymentMapper(result.reason.deploymentInfo, region, ctx);
+    if (!dbDeploy) {
+      return [
+        {
+          id: deploymentId,
+          status: 'KO',
+          message: 'Error getting updated deployment',
+        },
+      ];
+    }
+
+    // create the deploy in the database
+    await this.module.deployment.db.create(dbDeploy, ctx);
```

Comment:
You don't need this since there will be a `iasql_commit` after the RPC is finished. Also, this will prevent direct calls from other RPCs since all the FKs should be present.